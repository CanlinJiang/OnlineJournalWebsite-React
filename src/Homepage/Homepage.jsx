import React from 'react'
import {NavLink} from "react-router-dom"
import Frame from "../UtilitiesComponents/MainFrame/Frame"
import styles from "./Homepage.module.css"



export default function Homepage() {
    const Header = () => (
        <nav className={styles.topNav}>
            <ul className={styles.navigation}> 
                <li id = "home" ><NavLink to="/">Home</NavLink></li>
                <li id = "addNewArticle"><NavLink to="/addnewarticle">Add New Article</NavLink></li>
            </ul>
        </nav>
    )


    const Body = () => (
    <div className={styles.container}>
        <div className={styles.titleContainer}>
            <h1>Article collection</h1>
            <nav className={styles.midNav}>
            <ul className={styles.navigation}>
                <li ><NavLink to="/minelist">Mine's</NavLink></li>
                <li id ><NavLink to="/moodchoosing">Moods</NavLink></li>
                <li ><NavLink to="/randomwalk">Random Walk</NavLink></li>
            </ul>
            </nav>
        </div>
        <img className={styles.imageContainer}
            srcset={
            "images/IMG_0489-300_x_300.JPG 300w",
            "images/IMG_0489-600_x_600.JPG 600w",
            "images/IMG_0489-900_x_900.JPG 900w"
            }
            sizes="
            (max-width: 800px) 300px,
            (max-width: 1000px) 400px,
            (max-width: 1300px) 500px, 
            700px
            "
            src="IMG_0489-600_x_600.JPG"
            alt="black and white photo: afternoon">   
          </img>
    </div>
    )
        
    



    return (
        <Frame Header={Header} BodyComponent={Body}/>

    )
}
