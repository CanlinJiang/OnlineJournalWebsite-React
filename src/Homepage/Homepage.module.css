body {
    min-width: 330px;
}

.topNav ul {
    display: flex;
    width: 100%;
    justify-content: flex-end;
}

.midNav ul {
    font-size: 2vw;
    display: flex;
    flex-wrap: nowrap;
    width: 100%;
    justify-content: flex-start;
}

.midNav ul a{
    padding: 5px 15px 5px 0px;
}

.navigation {
    list-style-type: none;
    margin: 0;
    padding: 0;
}

.navigation a{
    display: block;
    color: black;
    text-align: center;
    text-decoration: none;
    padding: 14px 10px;  
}

.navigation a:hover{
    text-decoration: underline;
    
}

.container{
    display: flex;
    flex-wrap: nowrap;
    flex-direction: row;
}

.titleContainer{
    flex-grow: 1;
    align-self: center;
    margin-bottom: 2rem;
}
.titleContainer h1{
    font-size: 5vw;
    font-family: "prata",serif;
}

.imageContainer{
    flex-grow: 1;
}


@media (max-width: 767px) {
    .container{
        display: block;
    }
    .titleContainer{
        width: 100%;
        text-align: center;
    }
    .titleContainer h1{
        font-size: 2rem;
    }
    .midNav ul {
        font-size: small;
        justify-content: center;
    }
    .imageContainer{
        display: block;
        width: 100%;
    }
    .topNav ul {
        justify-content: center;
    }
}

@media (min-width: 768px) and (max-width: 1023px) {
    .container {
        margin-bottom: 200px;
    }
}
