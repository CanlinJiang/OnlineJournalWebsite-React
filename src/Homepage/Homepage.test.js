import Homepage from "./Homepage"
import React from "react";

describe("Homepage tests", () => {
    it("renders correctly", () => {
        
      const { container } = render(<Homepage />);
      expect(container).toMatchSnapshot();
    });
  
  });