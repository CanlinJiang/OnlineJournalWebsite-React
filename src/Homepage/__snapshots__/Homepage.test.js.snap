// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header"
    >
      <nav
        class="topNav"
      >
        <ul
          class="navigation"
        >
          <li
            id="home"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            id="addNewArticle"
          >
            <a
              href="/addnewarticle"
            >
              Add New Article
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="body"
    >
      <div
        class="container"
      >
        <div
          class="titleContainer"
        >
          <h1>
            Article collection
          </h1>
          <nav
            class="midNav"
          >
            <ul
              class="navigation"
            >
              <li>
                <a
                  href="/minelist"
                >
                  Mine's
                </a>
              </li>
              <li>
                <a
                  href="/moodchoosing"
                >
                  Moods
                </a>
              </li>
              <li>
                <a
                  href="/randomwalk"
                >
                  Random Walk
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <img
          alt="black and white photo: afternoon"
          class="imageContainer"
          sizes=" (max-width: 800px) 300px, (max-width: 1000px) 400px, (max-width: 1300px) 500px,  700px "
          src="IMG_0489-600_x_600.JPG"
          srcset="images/IMG_0489-900_x_900.JPG 900w"
        />
      </div>
    </div>
    <div
      class="footer"
    >
      <nav>
        <ul
          class="navigation"
        >
          <li
            id="aboutMe"
          >
            <a
              href="/aboutme"
            >
              About me
            </a>
          </li>
          <li
            id="contactMe"
          >
            <a
              href="/contactme"
            >
              Contact me
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
